lib/json_rails_logger/request_id_middleware.rb

```diff
@@ -11,7 +11,7 @@ def initialize(app)
     end
 
     def call(env)
-      request_id = env['action_dispatch.request_id']
+      request_id = env['HTTP_X_REQUEST_ID']
```

Comment:
If you look at the Rails code, following the link in the ticket, the header that is accessed is defined by `config.action_dispatch.request_id_header`. We can inspect this by opening the Rails console:

```text
ian@alder $ rails c
Running via Spring preloader in process 31057
Loading development environment (Rails 6.1.3)
irb(main):001:0> Rails.application.config.action_dispatch.request_id_header
=> "X-Request-Id"
```

So to correctly read the request ID from the header, this code should be either:

```ruby
request_id = env['X-Request-Id']
```

or (and this would be better, really):

```ruby
request_id = env[Rails.application.config.action_dispatch.request_id]
```

Reply:
It's definitely working currently, tested it using the command:
```
curl localhost:3000 -H "X-Request-Id: 058c70bdf4460415782d7cebb6a4390e"
```
but I'm happy to change it so it looks for `X-Request-Id` instead. The second option I think will still use the internally generated request id if none provided in the request, because previously it was `env['action_dispatch.request_id']`

Reply:
No, if you look at [the code I linked to](https://github.com/rails/rails/blob/5aaaa1630ae9a71b3c3ecc4dc46074d678c08d67/actionpack/lib/action_dispatch/middleware/request_id.rb), the internal request ID is only generated if the header is not present when calling the `request_id` method. Accessing the headers directly with `[]`, using any key, cannot call the internal request ID generator

Reply:
I have tried both options and none work for me, in both cases the `X-Request-Id` passed in the request header is not picked up in the middleware and ends up missing from the returning JSON. Further more, no internal request id is generated either. Are they working for you @ijdickinson ?